mne/decoding/receptive_field.py

```diff
@@ -0,0 +1,209 @@
+
+        self.model.fit(X_del.T, y.T)
+        self.msk_remove = msk
+
+    def transform(self, X=None, y=None):
+        """Does nothing, used for sklearn compatibility.
+
+        Parameters
+        ----------
+        X : None
+            Used for sklearn compatibility.
+        y : None
+            Used for sklearn compatibility.
+        """
+        return self
+
+    def fit_transform(self, X=None, y=None):
+        """Does nothing, used for sklearn compatibility.
+
+        Parameters
+        ----------
+        X : None
+            Used for sklearn compatibility.
+        y : None
+            Used for sklearn compatibility.
+        """
+        return self
+
+    def predict(self, X, y=None):
+        """Make predictions using a receptive field.
+
+        Parameters
+        ----------
+        X : array, shape ([n_epochs], n_channels, n_times)
+            The input features for the model.
+        y : None
+            Used for sklearn compatibility.
+
+        Returns
+        -------
+        y_pred : array, shape (n_times * n_epochs)
+            The output predictions with time concatenated.
+        """
+        X_del, msk = self._delay_for_fit(X)
+        X_del = X_del[..., ~msk]
+        X_del = X_del.reshape([-1, X_del.shape[-1]]).T
+        y_pred = self.model.predict(X_del)
+        return y_pred
+
+    def _delay_for_fit(self, X):
+        # First delay
+        X_del = delay_time_series(
+            X, self.lags, self.sfreq)
+
+        # Mask for removing edges later
+        msk_helper = np.ones(X.shape[-1])
+        msk_helper = delay_time_series(msk_helper, self.lags, self.sfreq)
+        msk = (msk_helper == 0).any(axis=0)
+        return X_del, msk
+
+    def plot_coefs(self, ax=None):
+        """Plot the fitted model coefficients.
+
+        Parameters
+        ----------
+        ax : instance of matplotlib axis object | None
+            The axis to use for plotting.
+
+        Returns
+        -------
+        ax : instance of matplotlib axis object
+            The axis used for plotting.
+        """
+        coefs = self.coefs
+
+        # Plot
+        if ax is None:
+            fig, ax = plt.subplots()
+        ax.pcolormesh(self.lags, self.stim_features,
+                      coefs, cmap='coolwarm')
+        _ = plt.setp(ax,
+                     xlim=[self.lags.min(), self.lags.max()],
+                     ylim=[self.stim_features.min(), self.stim_features.max()])
+        return ax
+
+    @property
+    def coefs(self):
+        """Return the model coefficients."""
+        coefs = get_coefs(self.model)
+        coefs = coefs.reshape([len(self.stim_features), len(self.lags)])
+        return coefs
+
+
+def delay_time_series(X, delays, sfreq=1.):
+    """Return a time-lagged input time series.
+
+    Parameters
+    ----------
+    X: array, shape ([n_epochs], n_features, n_times)
+        The time series to delay.
+    delays: array of floats, shape (n_delays,)
```

Comment:
what shape do you think it could be that I'm missing? In my mind it's just a space to put an array (or list) of delays to use